demo-circuit/src/naive_threshold_sig_w_key_rotation/constraints/data_structures.rs

```diff
@@ -175,12 +175,44 @@ impl ValidatorKeysUpdatesGadget {
         &self,
         mut cs: CS,
         new_validators_keys_leaves: &[FieldElementGadget],
+        epoch_id: &UInt32,
+        ledger_id: &FieldElementGadget,
     ) -> Result<(), SynthesisError> {
+
+        let secret_key_tag = UInt8::constant('s' as u8);
+        let master_key_tag = UInt8::constant('m' as u8);
+        let salt = UInt8::constant(VALIDATOR_HASH_SALT);
+
```

Comment:
epoch_id is always chosen at random in any test, so there should enough different values across different tests to ensure the consistency between the primitive and the gadget in case the tests are successful